recipes_source/recipes/profiler.py

```diff
@@ -0,0 +1,203 @@
+# model_inference            3.541ms         69.571ms    69.571ms      1
+# conv2d                     69.122us        40.556ms    2.028ms       20
+# convolution                79.100us        40.487ms    2.024ms       20
+# _convolution               349.533us       40.408ms    2.020ms       20
+# mkldnn_convolution         39.822ms        39.988ms    1.999ms       20
+# batch_norm                 105.559us       15.523ms    776.134us     20
+# _batch_norm_impl_index     103.697us       15.417ms    770.856us     20
+# native_batch_norm          9.387ms         15.249ms    762.471us     20
+# max_pool2d                 29.400us        7.200ms     7.200ms       1
+# max_pool2d_with_indices    7.154ms         7.170ms     7.170ms       1
+# -------------------------  --------------  ----------  ------------  ---------
+
+######################################################################
+# Here we see that, as expected, most of the time is spent in convolution (and specifically in ``mkldnn_convolution``
+# for PyTorch compiled with MKL-DNN support).
+# Note the difference between self cpu time and cpu time - operators can call other operators, self cpu time exludes time
+# spent in children operatator calls, while total cpu time includes it.
+#
+# To get a finer granularity of results and include operator input shapes, pass ``group_by_input_shape=True``:
+
+print(prof.key_averages(group_by_input_shape=True).table(sort_by="cpu_time_total", row_limit=10))
+
+# (omitting some columns)
+# -------------------------  -----------  --------  -------------------------------------
+# Name                       CPU total    # Calls         Input Shapes
+# -------------------------  -----------  --------  -------------------------------------
+# model_inference            69.571ms     1         []
+# conv2d                     9.019ms      4         [[5, 64, 56, 56], [64, 64, 3, 3], []]
+# convolution                9.006ms      4         [[5, 64, 56, 56], [64, 64, 3, 3], []]
+# _convolution               8.982ms      4         [[5, 64, 56, 56], [64, 64, 3, 3], []]
+# mkldnn_convolution         8.894ms      4         [[5, 64, 56, 56], [64, 64, 3, 3], []]
+# max_pool2d                 7.200ms      1         [[5, 64, 112, 112]]
+# conv2d                     7.189ms      3         [[5, 512, 7, 7], [512, 512, 3, 3], []]
+# convolution                7.180ms      3         [[5, 512, 7, 7], [512, 512, 3, 3], []]
+# _convolution               7.171ms      3         [[5, 512, 7, 7], [512, 512, 3, 3], []]
+# max_pool2d_with_indices    7.170ms      1         [[5, 64, 112, 112]]
+# -------------------------  -----------  --------  --------------------------------------
+
+
+######################################################################
+# 4. Use profiler to analyze memory consumption
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# PyTorch profiler can also show the amount memory allocated (freed) for the tensors
```

Comment:
What does allocated (freed) mean? Also, a brief explanation of what CPU Mem and Self CPU Mem is would be useful (I take it, similar to time, it's memory allocated by function and its children, or function itself).